CMakeLists.txt

```diff
@@ -309,6 +313,7 @@ zephyr_compile_options(${COMPILER_OPT_AS_LIST})
 
 # TODO: Include arch compiler options at this point.
 
+# TODO: special handling for mwdt here, mwdt is Clang based
```

Comment:
So mwdt is clang-based, meaning the parser.
But does it accept {all the same, a superset, or different} options as clang?
I'd suggest mwdt gets its own CMAKE_C_COMPILER_ID, unless it is completely commandline compatible with clang.

FYI, verbatim compiler options in the root CMakeLists is considered cheating :) Please consider using the abstractions already inplace.

Reply:
it's clang based, but lots of different options to keep compatible with old metware compiler.